File: my_publisher_package/subscriber_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import Float64MultiArray

class PoseSubscriber(Node):
    def __init__(self):
        super().__init__('pose_subscriber')
        self.subscription = self.create_subscription(
            Float64MultiArray,
            'omni_data',
            self.msg_callback,
            10)

    def msg_callback(self, msg):
        received_data = msg.data
        self.get_logger().info('Received data: "%s"' % msg.data)
        # Other processing if needed

def main(args=None):
    rclpy.init(args=args)
    pose_subscriber = PoseSubscriber()
    try:
        rclpy.spin(pose_subscriber)
    except KeyboardInterrupt:
        pass

    pose_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: my_publisher_package/subscriber_node.cpp
#include <iostream>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float64_multi_array.hpp"

class Float64ArraySubscriber : public rclcpp::Node
{
public:
  Float64ArraySubscriber()
      : Node("float64_array_subscriber")
  {
    subscription_ = this->create_subscription<std_msgs::msg::Float64MultiArray>(
        "omni_data", 10, std::bind(&Float64ArraySubscriber::callback, this, std::placeholders::_1));
  }

private:
  void callback(const std_msgs::msg::Float64MultiArray::SharedPtr msg) const
  {
    std::cout << "I heard: ";
    for (const auto& value : msg->data)
      std::cout << value << " ";
    std::cout << std::endl;
  }

  rclcpp::Subscription<std_msgs::msg::Float64MultiArray>::SharedPtr subscription_;
};

int main(int argc, char *argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<Float64ArraySubscriber>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
